config/docker/fragment/kselftest.jinja2

```diff
@@ -22,4 +22,7 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
    libasound2-dev{{ pkgarch }} \
    libasound2-dev \
    patch \
-   pkg-config
+   pkg-config \
+   clang \
```

Comment:
This seems to be rather counter to the idea of having the GCC container - it should be fixed at the build-config.yaml level where we decide which toolchains we're going to build kselftest with, if it just can't be fixed to work with GCC then we should probably just disable kselftest for RISC-V GCC builds

Reply:
note this isn't riscv specific.  clang is required to build this test for all arches.

Reply:
We should probably be skipping the test if we don't have clang then (really, fixing upstream to do so) shouldn't we?

Reply:
I agree with the fact having clang in a GCC container seems wrong. But I was thinking that it would be nice to have most of the kselftest testsuites built ok in a single package (BTW this issue is not RISCV specific: it happens for the other arches).
On the other side, I tried building kselftest with a "clang" docker and I can see there are a lot of failures. As a consequence, I don't think we can consider using clang (only) to build kselftest for now.

Reply:
Yes, clang has a lot of trouble with kselfest at the minute - I only just fixed it a release or two ago to actually try to build with clang when asked to but there's a whole bunch more work needed.

Reply:
I think the (short-term) solution here is to just leave out clang and accept that this test will not be run in KernelCI for any arches.  That's what is happening with the other arches.

@broonie your other idea of only building kselftsest with clang is a possibility, but currently we are only doing kselftest builds under gcc (managed by `build-configs.yaml`) but TBH, I don't know if anyone has done any work to be sure that all of kselftests build under clang.

Reply:
I did a very small amount of initial work on it, up until a couple of releases ago it used to just not even try to cross compile when using clang which just didn't work at all.

Reply:
I just dropped this part as discussed